include/xtensor/xstrides.hpp

```diff
@@ -102,7 +102,13 @@ namespace xt
     inline auto compute_size(const shape_type& shape) noexcept
     {
         using size_type = std::decay_t<typename shape_type::value_type>;
-        return std::accumulate(shape.cbegin(), shape.cend(), size_type(1), std::multiplies<size_type>());
+        auto end = shape.cend();
+        size_type result = 1;
+        for (auto it = shape.cbegin(); it != end; ++it)
+        {
+            result *= *it;
+        }
+        return result;
```

Comment:
Since the `compute_size` function is not constexpr, we can keep the `std::accumulate` algorithm for the moment.